ghcide/src/Development/IDE/Spans/AtPoint.hs

```diff
@@ -308,6 +311,24 @@ atPoint IdeOptions{} (HAR _ hf _ _ (kind :: HieKind hietype)) (DKMap dm km) env
             UnhelpfulLoc {} | isInternalName name || isSystemName name -> Nothing
             _ -> Just $ "*Defined " <> printOutputable (pprNameDefnLoc name) <> "*"
 
+        prettyInstances :: (Either ModuleName Name, IdentifierDetails hietype) -> IO (Maybe T.Text)
+        prettyInstances (Right n, _) =
+            fmap (wrapHaskell . T.unlines . fmap printOutputable) <$> instancesForName
+            where
+                instancesForName :: IO (Maybe [ClsInst])
+                instancesForName = runMaybeT $ do
+                    typ <- MaybeT . pure $ lookupNameEnv km n >>= tyThingToType
+                    liftIO $ evalGhcEnv env $ getInstancesForType typ
+
+                tyThingToType :: TyThing -> Maybe Type
```

Comment:
Name change since otherwise this is confusing with e.g. `safeTyThingType` which gets the type _of_ a `TyThing`
```suggestion
                -- | Gets the datatype 'Type' corresponding to a TyThing, if it represents a datatype or
                -- a data constructor
                tyThingAsDataType :: TyThing -> Maybe Type
```